keiko-redis/src/main/kotlin/com/netflix/spinnaker/q/redis/RedisQueue.kt

```diff
@@ -376,9 +383,34 @@ class RedisQueue(
   private fun JedisCommands.zismember(key: String, member: String) =
     zrank(key, member) != null
 
-  private fun Message.hash() =
+  private fun JedisCommands.anyZismember(key: String, members: Set<String>) =
+    members.any { zismember(key, it) }
+
+  private fun JedisCommands.firstFingerprint(key: String, fingerprint: Fingerprint) =
+    fingerprint.all.firstOrNull { zismember(key, it) }
+
+  @Deprecated("Hashes the attributes property, which is mutable")
+  private fun Message.hashV1() =
     Hashing
       .murmur3_128()
       .hashString(toString(), Charset.defaultCharset())
       .toString()
+
+  private fun Message.hashV2() =
+    hashObjectMapper.convertValue(this, MutableMap::class.java)
+      .apply { remove("attributes") }
+      .let {
+        Hashing
+          .murmur3_128()
+          .hashString("v2:${hashObjectMapper.writeValueAsString(it)}", Charset.defaultCharset())
```

Comment:
Wouldn't it be better to specify an explicit encoding here (e.g. UTF-8) than relying on the default charset which may change unexpectedly?

Reply:
Fair! Will change.